File: app/src/main/java/mm/pndaza/tipitakamyanmar/repository/CategoryRepository.java
package mm.pndaza.tipitakamyanmar.repository;

import android.database.Cursor;
import android.database.sqlite.SQLiteOpenHelper;

import java.util.ArrayList;

import mm.pndaza.tipitakamyanmar.model.Category;
import mm.pndaza.tipitakamyanmar.utils.SQLBuilder;

public class CategoryRepository extends BaseRepository {
    public CategoryRepository(SQLiteOpenHelper dbHelper) {
        super(dbHelper);
    }

    private static final String CATEGORY_TABLE = "category";
    private static final String CATEGORY_ID_COLUMN = "id";
    private static final String CATEGORY_NAME_COLUMN = "name";

    public ArrayList<Category> getAllCategories() {
        ArrayList<Category> categories = new ArrayList<>();
        String sql = new SQLBuilder().select(CATEGORY_ID_COLUMN, CATEGORY_NAME_COLUMN).from(CATEGORY_TABLE).build();
        try (Cursor cursor = getReadableDatabase().rawQuery(sql, null)) {
            if (cursor != null && cursor.moveToFirst()) {
                do {
                    int id = cursor.getInt(cursor.getColumnIndexOrThrow(CATEGORY_ID_COLUMN));
                    String name = cursor.getString(cursor.getColumnIndexOrThrow(CATEGORY_NAME_COLUMN));
                    categories.add(new Category(id, name));
                } while (cursor.moveToNext());
            }
        }
        return categories;
    }
}

File: app/src/main/java/mm/pndaza/tipitakamyanmar/repository/CategoryRepository.kt
package mm.pndaza.tipitakamyanmar.repository

import android.database.sqlite.SQLiteOpenHelper
import mm.pndaza.tipitakamyanmar.database.DatabaseSchema.CategoryTable
import mm.pndaza.tipitakamyanmar.model.Category

class CategoryRepository(dbHelper: SQLiteOpenHelper) : BaseRepository(dbHelper) {
    val allCategories: ArrayList<Category>
        get() {
            val categories =
                ArrayList<Category>()
            val sql = """
                SELECT ${CategoryTable.id}, ${CategoryTable.name}
                 FROM ${CategoryTable.table} ORDER BY ${CategoryTable.id} ASC
            """.trimIndent()
            readableDatabase.rawQuery(sql, null).use { cursor ->
                if (cursor.moveToFirst()) {
                    do {
                        val id =
                            cursor.getInt(cursor.getColumnIndexOrThrow(CategoryTable.id))
                        val name =
                            cursor.getString(cursor.getColumnIndexOrThrow(CategoryTable.name))
                        categories.add(Category(id, name))
                    } while (cursor.moveToNext())
                }
            }
            return categories
        }

}
